### PR TITLE
Move repository checkout to separated operation

### PR DESCRIPTION
**Pull request by webrouse**
Changed:
- Created `repository/checkout` operation, so in the next step we can add telemetry.

### CODE REVIEW

File: internal/pkg/api/server/templates/dependencies/dependencies.go [hunk 1]
```diff
@@ -141,20 +141,20 @@ func NewServerDeps(serverCtx context.Context, envs env.Provider, logger log.Pref
...
-		repositoryManager: repositoryManager,
+		Base:      baseDeps,
+		Public:    publicDeps,
+		serverCtx: serverCtx,
+		serverWg:  serverWg,
+		logger:    logger,
+	}
+
+	// Create repository manager
+	if v, err := repository.NewManager(serverCtx, defaultRepositories, d); err != nil {
```
Comment (webrouse) [inline internal/pkg/api/server/templates/dependencies/dependencies.go:154]: `NewManager` pouziva uz tiez `dependencies` interface.

File: internal/pkg/dependencies/mocked.go [hunk 2]
```diff
@@ -30,6 +31,7 @@ type mocked struct {
 	*project
 	envs                *env.Map
 	options             *options.Options
+	serverWg            *sync.WaitGroup
```
Comment (webrouse) [inline internal/pkg/dependencies/mocked.go:34]: Pridane `ServerWaitGroup()` do `dependencies.Mocked`.

File: internal/pkg/template/repository/manager.go [hunk 3]
```diff
@@ -73,27 +81,23 @@ func (m *Manager) ManagedRepositories() (out []string) {
...
-			ctx, cancel := context.WithTimeout(m.ctx, OperationTimeout)
-			defer cancel()
-
 			// Log start
 			startTime := time.Now()
 			m.logger.Infof(`checking out repository "%s:%s"`, repoDef.Url, repoDef.Ref)
 
 			// Checkout
-			repo, err := git.Checkout(ctx, repoDef.Url, repoDef.Ref, false, m.logger)
+			repo, err := checkoutOp.Run(ctx, repoDef, m.deps)
```
Comment (webrouse) [inline internal/pkg/template/repository/manager.go:95]: Kod presunuty do `checkout` operation./

File: pkg/lib/operation/repository/checkout/operation.go [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+)
+
+const Timeout = 30 * time.Second
+
+type dependencies interface {
+	Logger() log.Logger
+}
+
+func Run(ctx context.Context, repoDef model.TemplateRepository, d dependencies) (*git.Repository, error) {
+	// Create context with timeout
```
Comment (webrouse) [inline pkg/lib/operation/repository/checkout/operation.go:20]: Checkout operation, v dalsom kroku tu pribudne telemetria (DD span), aby sme to vedeli trackovat v APMku.

**Review by JakubMatejka — APPROVED** [review]
👍

**Pull request merged**